### DISCUSSION TITLE
Pooler restart when kubernetes api server upgraded

### DISCUSSION THREAD

**Discussion by dspeck1** [cloudnative-pg/cloudnative-pg#2708] [Q&A]
We did an upgrade of the kubernetes api servers (not the nodes) and the poolers/pgbouncer restarted.  We were not expecting that and thought they would stay up.  Is there a dependency between the poolers and kubernetes api connectivity?  The database pods stayed up.

**Comment by gbartolini** [comment]
That should not happen. By any chance, did you keep the logs of the operator and the pooler?
